squire_lib/src/ffi_tournament.rs

```diff
@@ -53,41 +55,74 @@ impl TournamentId {
         return ptr;
     }
 
+    /// Adds a player to a tournament
+    /// On error a NULL UUID is returned
+    #[no_mangle]
+    pub unsafe extern "C" fn tid_add_player(self: Self, __name: *const c_char) -> PlayerId {
+        let name: &str = unsafe { CStr::from_ptr(__name).to_str().unwrap() };
+        let op: TournOp = TournOp::RegisterPlayer(name.to_string());
+
+        match FFI_TOURNAMENT_REGISTRY.get_mut().unwrap().get_mut(&self) {
+            Some(mut t) => {
+                match t.apply_op(op) {
+                    Ok(RegisterPlayer(PlayerIdentifier::Id(id))) => {
+                        return id;
+                    }
+                    Err(t_err) => {
+                        println!("[FFI]: {t_err}");
+                        return PlayerId::new(Uuid::from_bytes(NULL_UUID_BYTES));
+                    }
+                    // We are known the from that the data will take if it exists
+                    // so we can ignore the other outcomes
+                    _ => {
+                        return PlayerId::new(Uuid::from_bytes(NULL_UUID_BYTES));
```

Comment:
I just double-checked this, and it does work. `Uuid` implements the `Default` trait, which is the same as the null bits. The following is a much more succinct way of getting a null `PlayerId` (or any typed id).
```suggestion
                        return Uuid::default().into();
```